internal/sops/azkv/keysource.go

```diff
@@ -115,7 +116,16 @@ func (key *MasterKey) EncryptIfNeeded(dataKey []byte) error {
 // Decrypt decrypts the EncryptedKey field with Azure Key Vault and returns
 // the result.
 func (key *MasterKey) Decrypt() ([]byte, error) {
-	c, err := azkeys.NewClient(key.VaultURL, key.token, nil)
+	creds := key.token
+	if key.token == nil {
+		// if there is no token credential set, use the DefaultAzureCredential
+		newCreds, err := azidentity.NewDefaultAzureCredential(nil)
```

Comment:
We still get the attempt to shell out to the CLI when `key.token` isn't set and we use the server from sops
https://github.com/mozilla/sops/blob/master/azkv/keysource.go#L142

Bu using the NewDefaultAzureCredentian, we attempt all the other methods(including Workload Identity) before trying to shell out

Reply:
Given that there are no CLIs in the controller container, this only confuses people. Locally if you run the tests on dev machine, it will find  `az`, but it will always fail in-cluster.